ts/sagas/watchDeferredActionsSaga.ts

```diff
@@ -0,0 +1,20 @@
+import { Effect, put, select, takeEvery } from "redux-saga/effects";
+import { DEFER_TO_LOGIN } from "../store/actions/constants";
+import { DeferToLogin, pushToDeferredActions } from "../store/actions/deferred";
+import { isPinLoginValidSelector } from "../store/reducers/pinlogin";
+
+function* handleDeferredActionSaga({
+  payload: action
+}: DeferToLogin): IterableIterator<Effect> {
+  const isPinValid: boolean = yield select(isPinLoginValidSelector);
+
+  if (isPinValid) {
+    yield put(action);
```

Comment:
I don't understand this - very hard to follow - if PIN is valid the action gets triggered right away? if not it gets deferred?

Reply:
If the pin is valid, then the user is logged in, so the action will be dispatched right away. Otherwise, the action will be dispatched as soon as the user is logged in.